plugins/woocommerce-admin/client/products/use-product-helper.ts

```diff
@@ -0,0 +1,213 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useDispatch } from '@wordpress/data';
+import { useCallback } from '@wordpress/element';
+import {
+	Product,
+	ProductsStoreActions,
+	ProductStatus,
+	PRODUCTS_STORE_NAME,
+	ReadOnlyProperties,
+	productReadOnlyProperties,
+} from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+import { navigateTo } from '@woocommerce/navigation';
+
+function removeReadonlyProperties(
+	product: Product
+): Omit< Product, ReadOnlyProperties > {
+	productReadOnlyProperties.forEach( ( key ) => delete product[ key ] );
+	return product;
+}
+
+function getNoticePreviewActions( status: ProductStatus, permalink: string ) {
+	return status === 'publish' && permalink
+		? [
+				{
+					label: __( 'View in store', 'woocommerce' ),
+					onClick: () => {
+						recordEvent( 'product_preview_changes', {
+							new_product_page: true,
+						} );
+						window.open( permalink, '_blank' );
+					},
+				},
+		  ]
+		: [];
+}
+
+export function useProductHelper() {
```

Comment:
This is a custom hook I created to move some of the specific create, copy, update, delete logic so the `ProductFormActions` component doesn't get super long.
This would also be helpful in the case we need to call these functions from any other component (for example when we create the new product list page at some point).